extra/html-extra/HtmlExtension.php

```diff
@@ -124,4 +130,77 @@ public static function htmlCva(array|string $base = [], array $variants = [], ar
     {
         return new Cva($base, $variants, $compoundVariants, $defaultVariant);
     }
+
+    public static function htmlAttrMerge(...$arrays): array
+    {
+        $result = [];
+
+        foreach ($arrays as $argNumber => $array) {
+            if (!$array) {
+                continue;
+            }
+
+            if (!is_iterable($array)) {
+                throw new RuntimeError(sprintf('The "attr_merge" filter only works with arrays or "Traversable", got "%s" for argument %d.', \gettype($array), $argNumber + 1));
+            }
+
+            $array = CoreExtension::toArray($array);
+
+            foreach (['class', 'style', 'data', 'aria'] as $deepMergeKey) {
+                if (isset($array[$deepMergeKey])) {
+                    $value = $array[$deepMergeKey];
+                    unset($array[$deepMergeKey]);
+
+                    if (!is_iterable($value)) {
+                        $value = (array) $value;
+                    }
+
+                    $value = CoreExtension::toArray($value);
+
+                    $result[$deepMergeKey] = array_merge($result[$deepMergeKey] ?? [], $value);
+                }
+            }
+
+            $result = array_merge($result, $array);
+        }
+
+        return $result;
+    }
+
+    public static function htmlAttr(Environment $env, ...$args): string
+    {
+        $attr = self::htmlAttrMerge(...$args);
+
+        if (isset($attr['class'])) {
+            $attr['class'] = trim(implode(' ', $attr['class']));
+        }
+
+        if (isset($attr['style'])) {
+            $style = '';
+            foreach ($attr['style'] as $name => $value) {
+                if (is_numeric($name)) {
+                    $style .= $value.'; ';
+                } else {
+                    $style .= $name.': '.$value.'; ';
+                }
```

Comment:
I'm not sure why this specific case for style.. i may have missed the reason earlier in the discussion ?

Reply:
In case `style` is something like `['color: red', 'font-weight: bold']`.

Reply:
I have two other questions then (sorry)

What if  "aria" is something like `['aria-disabled', 'aria-current']` ?

Does that mean i can do something like this ?

```twig
{{ html_attr(
   { style: ['color: red;'] },
   { style: {'color': 'blue'} },
   { style: {'color': 'green'} },
   { style: ['', 'color: black;'] },
) }}
```

I'm shared between "i understand each of these usages" and "this could be very un-predictable when/if args are assembled from different layers / templates :|

Reply:
Yes, definetly many edge cases I haven't thought through. Make sure you don't miss the inital idea in #3907, fwiw.

I think we should aim for behaviour similar to `array_merge` and _not_ try to do any kind of parsing/interpretation. For example:

* `html_attr_merge( { style: {'color': 'blue'} }, { style: {'color': 'green'} } } ) }}` would be `{ style: {'color': 'green'} }`
* `html_attr_merge( { style: ['color: blue'] }, { style: { ['color: green']} } } ) }}` would be `{ style: ['color: blue', 'color: green'] }` – if you want "real" overrides, use semantic keys
* `html_attr_merge( { aria: { 'aria-label': 'that' }, {'label': 'this'} } )` would be `{ 'aria-label': 'this' }` – I'd rewrite keys from the `aria` and `data` sub-array to the "flat" values first and then merge those.

I hope I can come up with reasonable test cases for all those situations so we can discuss them before merge.